core/src/test/scala/integration/kafka/network/DynamicConnectionQuotaTest.scala

```diff
@@ -100,8 +100,7 @@ class DynamicConnectionQuotaTest extends BaseRequestTest {
       val socket = connect("PLAINTEXT")
       socket.setSoTimeout(1000)
       try {
-        val response = sendAndReceive(produceRequest, ApiKeys.PRODUCE, socket)
-        assertEquals(0, response.remaining)
```

Comment:
Why don't we need this assertion anymore?

Reply:
The assertion was actually never getting executed. The call to `sendAndReceive` raises a timeout exception, which is what `verifyMaxConnections` is looking for.